File: Titan Backend/src/main/java/com/example/demo/service/OrderService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.model.Order;

public interface OrderService 
{
   public Order saveOrder(Order order);
   public Order getOrderById(long id);
   public List<Order> getOrders();
   public Order updateOrder(Order order);
   public void deleteOrder(long id);
}

File: Titan Frontend/src/components/OrderService.js
import axios from 'axios';

const API_BASE_URL = "http://localhost:8080"; // Ensure this matches your Spring Boot server URL

class OrderService {

    saveOrder(order) {
        return axios.post(`${API_BASE_URL}/order`, order);
    }

    getOrderById(orderId) {
        return axios.get(`${API_BASE_URL}/order/${orderId}`);
    }

    getOrders() {
        return axios.get(`${API_BASE_URL}/orders`);
    }

    updateOrder(orderId, orderData) {
        return axios.put(`${API_BASE_URL}/order/${orderId}`, orderData);
    }

    deleteOrder(orderId) {
        return axios.delete(`${API_BASE_URL}/order/${orderId}`);
    }
}

export default new OrderService();
